internal/storage/storage_handle.go

```diff
@@ -57,16 +57,20 @@ func createGRPCClientHandle(ctx context.Context, clientConfig *storageutil.Stora
 	}
 
 	var clientOpts []option.ClientOption
-	tokenSrc, err := storageutil.CreateTokenSource(clientConfig)
-	if err != nil {
-		err = fmt.Errorf("while fetching tokenSource: %w", err)
-		return
-	}
-	clientOpts = append(clientOpts, option.WithTokenSource(tokenSrc))
 
 	// Add Custom endpoint option.
 	if clientConfig.CustomEndpoint != nil {
 		clientOpts = append(clientOpts, option.WithEndpoint(clientConfig.CustomEndpoint.String()))
+		// Explicitly disable no-auth in case of custom-endpoint, aligned with the http-client.
+		// TODO: to revisit here when supporting TPC for grpc client.
+		clientOpts = append(clientOpts, option.WithoutAuthentication())
```

Comment:
In case of http, we create custom http client and token source goes with the custom [oauth2.Transport](https://github.com/GoogleCloudPlatform/gcsfuse/blob/0109379a2cf93918ed0f74c3f4166efb378466fc/internal/storage/storageutil/client.go#L86).

We can't do similar customization. Hence, we use option.WithTokenSource to set the tokenSource. And it fails because of [this](https://github.com/googleapis/google-api-go-client/blob/3daccecaae7b1bcd1371a44cb2d25b2fec4f34f2/internal/settings.go#L109) check irrespective of real or fake tokenSrc.

Although, I have agenda to discuss about [grpc.WithInsecure](https://github.com/googleapis/google-cloud-go/blob/67d353beefe3b607c08c891876fbd95ab89e5fe3/storage/grpc_client.go#L96) - like, what's the impact of generic custom-endpoint user?

Reply:
Resolving this, based on offline discussion:
```
1. We know the reason behind, why static token source is not working for gRPC client.
2. We will have a separate discussion for withInsecure, which will come into the picture while supporting TPC.
```